python/grass/jupyter/interactivemap.py

```diff
@@ -346,6 +347,95 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def setup_drawing_interface(self):
+        """
+        Allow Users to draw Geometry and Save/Delete them.
+        """
+        import ipywidgets as widgets  # pylint: disable=import-outside-toplevel
+
+        draw_control = self._ipyleaflet.DrawControl(edit=False, remove=False)
+        drawn_geometries = []
+        geo_json_layers = {}
+        save_button_control = None
+
+        def handle_draw(_target, action, geo_json):
+            if action == "created":
+                drawn_geometries.append(geo_json)
+                print(f"Geometry created: {geo_json}")
+
+        draw_control.on_draw(handle_draw)
+
+        draw_button = widgets.ToggleButton(
+            icon="pencil",
+            value=False,
+            tooltip="Click to draw geometries",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        def toggle_geometry(change):
+            if change["new"]:
+                self.map.add_control(draw_control)
+                show_interface()
+            else:
+                drawn_geometries.clear()
+                self.map.remove_control(draw_control)
+                hide_interface()
+
+        draw_button.observe(toggle_geometry, names="value")
+
+        def show_interface():
+            nonlocal save_button_control
+
+            name_input = widgets.Text(
+                description="Name:",
+                style={"description_width": "initial"},
+                layout=widgets.Layout(
+                    width="50%",
+                    margin="1px 1px 1px 1px",
+                ),
+            )
+            save_button = widgets.Button(
+                description="Save Geometries",
+                layout=widgets.Layout(width="50%", margin="1px 1px 1px 1px"),
+            )
+
+            def save_geometries(_b):
+                name = name_input.value
+                if name and drawn_geometries:
+                    for geometry in drawn_geometries:
+                        geometry["properties"]["name"] = name
+                    geo_json = {
+                        "type": "FeatureCollection",
+                        "features": drawn_geometries,
+                    }
+                    save_vector(name, geo_json)
+                    geo_json_layer = self._ipyleaflet.GeoJSON(data=geo_json, name=name)
+                    geo_json_layers[name] = geo_json_layer
+                    self.map.add_layer(geo_json_layer)
```

Comment:
When you add the drawn geometries as geojson layer, it should remove the drawings, otherwise it shows up twice.